### PR TITLE
Don't disable logging if it's needed elsewhere

### PR DESCRIPTION
**Pull request by permcody**
PerformanceData postprocessor needs access to the perf log even if it's not being printed to the screen.

### CODE REVIEW

File: framework/src/outputs/Console.C [hunk 1]
```diff
@@ -251,6 +236,24 @@ Console::~Console()
...
+  {
+    if (_perf_log || _setup_log || _solve_log || _perf_header || _setup_log_early)
+      _app.setLoggingRequested();
+
+    // Disable performance logging if nobody needs logging
+    if (!_app.getLoggingRequested())
+      Moose::perf_log.disable_logging();
+
+    // Disable libMesh log
+#ifdef LIBMESH_ENABLE_PERFORMANCE_LOGGING
```
Comment (jwpeterson) [inline framework/src/outputs/Console.C]: I realize you didn't add this code block in this PR, but I don't think it needs to be wrapped in the `ifdef`.  The global `perflog` object exists no matter what, the `ifdef` only controls the behavior of the `START_LOG`, `STOP_LOG`, etc. macros.

File: test/tests/postprocessors/print_perf_data/tests [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
     check_files = print_perf_data_out.csv
   [../]
+
+  [./use_log_data_no_print]
+    type = RunApp
+    input = use_log_data_no_print.i
+    # Make sure that the perf_log table doesn't just report
+    # the same active time each timestep
+    # This RE looks for a time on subsequent lines that are identical
+    absent_out = '(\s(\S+)\s\|\n)[^\n]*\1'
```
Comment (jwpeterson) [inline test/tests/postprocessors/print_perf_data/tests]: As a regex-challenged person, I don't really understand why this regex works.  Can you maybe give an example in the comments of the output you are trying to make sure is absent?

**Review by jwpeterson — CHANGES_REQUESTED** [review]

File: framework/include/base/MooseApp.h [hunk 3]
```diff
@@ -468,6 +468,12 @@ class MooseApp :
   /// Returns whether the Application is running in check input mode
   bool checkInput() const { return _check_input; }
 
+  /// Sets a Boolean indicating that at least one object is requesting performance logging in this application
+  void setLoggingRequested() { _logging_requested = true; }
+
+  /// Returns a Boolean indicating whether performance logging is requested in this application
+  bool getLoggingRequested() const { return _logging_requested; }
+
```
Comment (aeslaughter) [inline framework/include/base/MooseApp.h]: Just to reduce adding more stuff to MooseApp, should these just go into OutputWarehouse? The Console and PerformanceData have access to the OutputWarehouse from _app, but if you prefer this I don't mind that much.

File: framework/src/outputs/Console.C (hunk 1 above)
Comment (permcody) [inline framework/src/outputs/Console.C]: I'll strip all these in another commit.

File: test/tests/postprocessors/print_perf_data/tests (hunk 2 above)
Comment (permcody) [inline test/tests/postprocessors/print_perf_data/tests]: 👍

File: framework/include/base/MooseApp.h (hunk 3 above)
Comment (permcody) [inline framework/include/base/MooseApp.h]: will do

**Review by aeslaughter — APPROVED** [review]

**Review by jwpeterson — APPROVED** [review]

**Pull request merged**